File: java/revenj-core/src/main/java/org/revenj/database/postgres/PostgresBuffer.java
package org.revenj.database.postgres;

public interface PostgresBuffer {

	char[] getTempBuffer();

	void initBuffer();

	void initBuffer(char c);

	void addToBuffer(char c);

	void addToBuffer(char[] c);

	void addToBuffer(char[] c, int len);

	void addToBuffer(char[] c, int start, int emd);

	void addToBuffer(String input);

	String bufferToString();
}

File: scala/revenj-core/src/main/scala/net/revenj/database/postgres/PostgresBuffer.scala
package net.revenj.database.postgres

trait PostgresBuffer {
  def tempBuffer: Array[Char]

  def initBuffer(): Unit

  def initBuffer(c: Char): Unit

  def addToBuffer(c: Char): Unit

  def addToBuffer(c: Array[Char]): Unit

  def addToBuffer(c: Array[Char], len: Int): Unit

  def addToBuffer(c: Array[Char], start: Int, emd: Int): Unit

  def addToBuffer(input: String): Unit

  def bufferToString(): String
}
